store/storetest/team_store.go

```diff
@@ -678,6 +681,89 @@ func testGetAllPrivateTeamPageListing(t *testing.T, ss store.Store) {
 	}
 }
 
+func testGetAllPublicTeamPageListing(t *testing.T, ss store.Store) {
+	o1 := model.Team{}
+	o1.DisplayName = "DisplayName"
+	o1.Name = "z-z-z" + model.NewId() + "b"
+	o1.Email = MakeEmail()
+	o1.Type = model.TEAM_OPEN
+	o1.AllowOpenInvite = true
+	_, err := ss.Team().Save(&o1)
+	require.Nil(t, err)
+
+	o2 := model.Team{}
+	o2.DisplayName = "DisplayName"
+	o2.Name = "zz" + model.NewId() + "b"
+	o2.Email = MakeEmail()
+	o2.Type = model.TEAM_OPEN
+	o2.AllowOpenInvite = false
+	_, err = ss.Team().Save(&o2)
+	require.Nil(t, err)
+
+	o3 := model.Team{}
+	o3.DisplayName = "DisplayName"
+	o3.Name = "z-z-z" + model.NewId() + "b"
+	o3.Email = MakeEmail()
+	o3.Type = model.TEAM_INVITE
+	o3.AllowOpenInvite = true
+	_, err = ss.Team().Save(&o3)
+	require.Nil(t, err)
+
+	o4 := model.Team{}
+	o4.DisplayName = "DisplayName"
+	o4.Name = "zz" + model.NewId() + "b"
+	o4.Email = MakeEmail()
+	o4.Type = model.TEAM_INVITE
+	o4.AllowOpenInvite = false
+	_, err = ss.Team().Save(&o4)
+	require.Nil(t, err)
+
+	if teams, listErr := ss.Team().GetAllPublicTeamPageListing(0, 10); listErr != nil {
+		t.Fatal(listErr)
+	} else {
+		for _, team := range teams {
+			require.Equal(t, team.AllowOpenInvite, true, "should have returned team with AllowOpenInvite as true")
+		}
+
+		if len(teams) > 10 {
```

Comment:
This feels kind of brittle. We've only made 4 teams so is there a risk that the database already has a bunch in it? Should we start the test be purging all teams if that doesn't happen automatically?

Reply:
@grundleborg, I created a new function that can be called to purge previous database results for teams.  I had brought this up in discussion with Joram that I think db clean up in tests could make for a decent HW campaign.  We've got quite a few tests that won't pass at the sub-test level because their comparison is based on the parent test function.  I feel like all sub tests should pass when run individually.  Regardless, that is for a separate discussion.  I'll have these changes checked in soon!

Reply:
Yeah, definitely agree with this and that a help wanted campaign could be good here. Our tests can be a lot more robust then they are now in many places.